libs/components/src/next/search-filter/vl-search-filter.component.ts

```diff
@@ -0,0 +1,149 @@
+            }
+        });
+        const buttonContainer = this.form?.querySelector<HTMLElement>(
+            'div:has(vl-button-next[type="submit"],button[type="submit"])'
+        );
+        buttonContainer?.classList.add('vl-search-filter-next--submit');
+
+        // Deze stylesheet moet toegevoegd worden aan de adoptedStyleSheets van het document
+        // omdat deze styles betrekking hebben op de slotted content en dus niet op de shadow dom
+        document.adoptedStyleSheets = [
+            ...document.adoptedStyleSheets,
+            searchFilterGlobalStyles.styleSheet as CSSStyleSheet,
+        ];
+    }
+
+    protected updated(changedProperties: PropertyValues) {
+        super.updated(changedProperties);
+
+        if (changedProperties.has('alt')) {
+            if (this.alt) {
+                this.form?.classList.add('vl-search-filter-next--form__alt');
+            } else {
+                this.form?.classList.remove('vl-search-filter-next--form__alt');
+            }
+        }
+        if (changedProperties.has('mobileModal')) {
+            if (this.mobileModal) {
+                this.buttonContainer?.classList.add('vl-search-filter-next--footer-modal');
+                this.form?.classList.add('vl-search-filter-next--form__mobile-modal');
+            } else {
+                this.buttonContainer?.classList.remove('vl-search-filter-next--footer-modal');
+                this.form?.classList.remove('vl-search-filter-next--form__mobile-modal');
+            }
+        }
+    }
+
+    render() {
+        const classes = {
+            'vl-search-filter-next': true,
+            'vl-search-filter-next--mobile-modal': this.mobileModal,
+        };
+        return html`
+            <div class=${classMap(classes)}>
+                ${!this.mobileModal ? this.renderTitle() : this.renderMobileModal()}
+                <div class="form-container">
+                    <slot></slot>
+                </div>
+            </div>
+        `;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.resizeObserver = new ResizeObserver((entries) => {
+            for (const entry of entries) {
+                this.mobileModal = entry.contentRect.width < 767;
+            }
+        });
+        this.resizeObserver.observe(document.body);
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        this.form?.removeEventListener('submit', this.handleSubmitModal);
+        if (this.resizeObserver) {
+            this.resizeObserver.disconnect();
+            this.resizeObserver = null;
+        }
+    }
+
+    private renderTitle() {
+        return this.filterTitle ? html`<p class="vl-search-filter-next--intro">${this.filterTitle}</p>` : nothing;
+    }
+
+    private renderMobileModal() {
+        return html`
+            <div class="vl-search-filter-next--header-modal">
+                <h2>${this.mobileModalTitle || this.filterTitle || 'Filter'}</h2>
```

Comment:
Is het voor deze h2?  Waarom dan niet gewoon een specifieke font-size voorzien voor een h2 in een div met de class vl-search-filter-next--header-modal ?